Source/WebCore/style/StyleTreeResolver.cpp

```diff
@@ -955,7 +955,10 @@ auto TreeResolver::updateStateForQueryContainer(Element& element, const RenderSt
         return QueryContainerAction::Continue;
     }
 
-    if (style->containerType() != ContainerType::Normal || previousContainerType != ContainerType::Normal) {
+    if (style->containerType() != ContainerType::Normal || (existingStyle && existingStyle->containerType() != ContainerType::Normal)) {
+        // If any of the queries use font-size relative units then a font size change may affect their evaluation.
+        if (existingStyle && style->computedFontSize() != existingStyle->computedFontSize())
+            descendantsToResolve = DescendantsToResolve::All;
```

Comment:
Would it be useful to store whether a container query has font relative units to avoid doing unnecessary work if the container query doesn't have any?

Reply:
That's a possible optimization but in practice the case is rare. Also normally changing an inherited property (like font properties) will cause all descendants to be recomputed in any case. The test case specifically stops this.